### PR TITLE
Porting nginx config reloader to python3

### PR DESCRIPTION
**Pull request by thebaron2**

### CODE REVIEW

**Comment by vdloo** [comment]
we also need to change the Debian packaging to python3, see https://github.com/ByteInternet/nginx_config_reloader/blob/master/debian/control#L7

for an example of how to upgrade that look at these:
https://github.com/ByteInternet/hypernode-kamikaze/blob/master/debian/control vs https://github.com/ByteInternet/hypernode-kamikaze3/blob/master/debian/control
and
https://github.com/ByteInternet/hypernode-kamikaze/blob/master/setup.py vs https://github.com/ByteInternet/hypernode-kamikaze3/blob/master/setup.py
etc etc

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: python
 python:
-  - 2.7
+  - 3.5
```
Comment (farzadghanei) [inline .travis.yml]: It's nice to keep the project moving to Python 3, but not dropping Python 2 support at once. so I suggest we keep `2.7` here to run tests with both `2.7` and `3.5`

File: tests/test_get_pid.py [hunk 2]
```diff
@@ -6,19 +6,19 @@
 class TestGetPid(unittest.TestCase):
 
     def test_that_get_pid_returns_pid_from_pidfile(self):
-        with mock.patch('__builtin__.open', mock.mock_open(read_data='42')):
+        with mock.patch('builtins.open', mock.mock_open(read_data='42')):
```
Comment (farzadghanei) [inline tests/test_get_pid.py]: I don't think this would work properly on Python 2.

**Comment by farzadghanei** [comment]
@thebaron2 it would be nice if this PR brings Python 3 compatibility instead of just dropping Python 2.

**Pull request merged**